test: emergency patch to repair the Windows CI

compnerd opened the pull request:
The removal of a key not in the set is an error.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd committed:
test: emergency patch to repair the Windows CI

The removal of a key not in the set is an error.

compnerd commented:
CC: @yln

compnerd commented:
@swift-ci please smoke test and merge

yln commented:
Thanks for cleaning up after me. Did not realize that remove throws if value is not present.

yln commented on test/lit.cfg:
```diff
@@ -1500,7 +1500,8 @@ else:
     else:
         # TSan runtime requires libdispatch on non-Apple platforms
-        config.available_features.remove('tsan_runtime')
+        if 'tsan_runtime' in config.available_features:
```
We can use `discard` instead of `remove` (does not throw if not present).

compnerd replied:
Good idea, I'll update this.

compnerd committed:
lit: prefer `discard` over `remove`

Use `discard` rather than `remove` so that the exception is not raised
and we do not need to check if the key is present.

compnerd commented:
@swift-ci please smoke test and merge

compnerd commented:
@swift-ci please smoke test and merge

compnerd commented:
@swift-ci please smoke test Linux platform

compnerd commented:
@swift-ci please smoke test Linux platform
